front/app/containers/IdeasNewPage/IdeasNewSurveyForm/SurveyHeading.tsx

```diff
@@ -192,6 +194,12 @@ const SurveyHeading = ({
                     buttonStyle={loggedIn ? 'primary' : 'delete'}
                     width="100%"
                     mb={isSmallerThanPhone ? '16px' : undefined}
+                    onClick={() => {
+                      // We need to invalidate any previously cached draft idea
+                      queryClient.invalidateQueries({
+                        queryKey: ideasKeys.item({ id: phaseId }),
+                      });
+                    }}
```

Comment:
Moved the cache invalidation here, since if this happens while we're "in" the survey (I.e. in the useUpdateIdea hook) it causes problems. In this case though, the invalidation happens once the user has seen + confirmed they want to leave the current instance of the survey.

Reply:
I'll add a comment in the code - yeah agreed that it's a bit odd. If it's in the useUpdateHook though with the other invalidations it cases problems while filling in the survey :thinking: